compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -101,10 +101,13 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
       if (name0.charAt(name0.length()-1) != ' ') name0
       else name0.substring(0, name0.length()-1)
 
-    /* We have to special-case fields of Ref types (IntRef, ObjectRef, etc.)
-     * because they are emitted as private by our .scala source files, but
-     * they are considered public at use site since their symbols come from
-     * Java-emitted .class files.
+    /* Java-defined fields are always accessed as if they were private. This
+     * is necessary because they are defined as private by our .scala source
+     * files, but they are considered `!isPrivate` at use site, since their
+     * symbols come from Java-emitted .class files. Fortunately, we can
+     * easily detect those as `isJavaDefined`. This includes fields of Ref
+     * types (IntRef, ObjectRef, etc.) which were special-cased at use-site
+     * in Scala.js < 0.6.15.
```

Comment:
Please add a comment here that this makes changing the length of the superclass chain of a .java defined Scala class a binary incompatible change.